File: src/main/java/com/design/platform/resourceplatform/interfaces/AdminController.java
package com.design.platform.resourceplatform.interfaces;


import com.design.platform.resourceplatform.entities.Account;
import com.design.platform.resourceplatform.services.AdminService;
import com.design.platform.resourceplatform.transfer.AdminBooth;
import com.design.platform.resourceplatform.transfer.AdminRecorder;
import com.design.platform.resourceplatform.utils.PageHolder;
import com.design.platform.resourceplatform.utils.PageParam;
import com.design.platform.resourceplatform.utils.PageUtilsKt;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/admin")
public class AdminController {

    private final AdminService service;

    public AdminController(AdminService service) {
        this.service = service;
    }

    // Methods GET
    // ===============================================

    @GetMapping
    public PageHolder<AdminBooth> GetList(PageParam param) {
        return service.GetAdminBoothList(PageUtilsKt.getAuto(param));
    }

    @GetMapping("/{id}")
    public AdminBooth GetAdmin(@PathVariable int id) {
        return service.GetAdminBooth(id);
    }

    // Methods POST
    // ===============================================

    @PostMapping
    public void UpdateAdmin(@RequestBody AdminRecorder recorder) {
        System.out.println(recorder);
        Account account =  service.UpdateAdmin(recorder);
        System.out.println(account);
    }

    @PutMapping
    public void CreateAdmin(@RequestBody AdminDefiner definer) {
        System.out.println(definer);
        Account account =  service.CreateAdmin(definer);
        System.out.println(account);
    }

    @DeleteMapping("/{id}")
    public void DeleteAdmin(@PathVariable int id){
        System.out.println(id);
        service.DestroyAdmin(id);
        System.out.println("done");
    }
}

File: src/main/java/com/design/platform/resourceplatform/interfaces/AdminController.kt
package com.design.platform.resourceplatform.interfaces

import com.design.platform.resourceplatform.security.Expression
import com.design.platform.resourceplatform.services.AdminService
import com.design.platform.resourceplatform.transfer.AdminDefiner
import com.design.platform.resourceplatform.transfer.AdminRecorder
import com.design.platform.resourceplatform.utils.PageParam
import com.design.platform.resourceplatform.utils.auto
import org.springframework.security.access.prepost.PostAuthorize
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/admin")
open class AdminController(
    open val service: AdminService
) {

    // Methods GET
    // ===============================================
    @GetMapping
    @PreAuthorize("@expression.isAdmin(principal)")
    open fun findAll(param: PageParam) = service.findAll(param.auto)

    @GetMapping("/{id}")
    @PreAuthorize("@expression.isAdmin(principal)")
    fun findOne(@PathVariable id: Int) = service.findOne(id)

    // Methods POST
    // ===============================================
    @PostMapping
    @PreAuthorize("@expression.isAdmin(principal)")
    fun update(@Validated @RequestBody recorder: AdminRecorder) {
        println(recorder)
        service.update(recorder)
    }

    @PutMapping
    @PreAuthorize("@expression.isAdmin(principal)")
    fun create(@Validated @RequestBody definer: AdminDefiner) {
        service.create(definer)
    }

    @DeleteMapping("/{id}")
    @PreAuthorize("@expression.isAdmin(principal)")
    fun delete(@PathVariable id: Int) {
        service.delete(id)
    }
}
